lib/src/klio/transforms/_helpers.py

```diff
@@ -260,9 +283,12 @@ def process(self, kmsg):
         item_path = self._get_absolute_path(item)
         item_exists = self.exists(item_path)
 
-        state = DataExistState.FOUND
         if not item_exists:
+            self.not_found_ctr.inc()
```

Comment:
Is there any way that `_KlioOutputDataMixin` can handle the metrics increments?  Usually I prefer the pattern that whatever creates the counter also is responsible for incrementing it (though it's not too out-of-place here either).

Reply:
The `_KlioOutputDataMixin` doesn't have a `process` method; if we were to add a `process` method to increment counters in the Mixins (and therefore calling `super(...).process(kmsg)` here), it wouldn't know which counter to increment (found or not found), unless we were to move the entire logic of this `process` method up into both `_KlioOutputDataMixin` and `_KlioInputDataMixin`. I get your argument, just curious if you have a preference to follow the creator -> incrementor rule, or keep to DRY. I'm a bit ambivalent so opinions welcome 🙂

Reply:
yeah I'm not sure there's a way to do it without making a bunch of other changes, so I'm fine with it as is.  It would be a bigger deal if we expected a lot of other subclasses or subclasses written outside of Klio, but I don't think either will be the case for a while.

Reply:
fair enough 👍🏻 thanks!